GuiApp/coreapplication.h

```diff
@@ -39,8 +41,57 @@ class CoreApplication : public QApplication
         if (event->type() == QEvent::FileOpen) {
             QFileOpenEvent *openEvent = static_cast<QFileOpenEvent *>(event);
             mFileToLoad = openEvent->file();
+            mMainWindow->loadFile(openEvent->file());
         }
 
         return QApplication::event(event);
     }
+
+    int exec() {
```

Comment:
Why does this rather hefty function implementation have to be in the header file?